### ISSUE TITLE
BindRuntimeType is not working correctly for FilterType unless bound in the filter convention (in 13 previews, worked in 12)

### ISSUE THREAD

**Issue by digitalpowers** [ChilliCream/graphql-platform#5262]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Talked to Pascal about this on the slack as well.

In 13-preview.28 i get errors "The type of member UnsignedIntFiled of the declaring type TestType is unknown" where it used to be known by just called .BindRuntimeType<uint, UnsignedIntType>() during setup. 

I did find that adding calls to .BindRuntimeType<uint, UnsignedIntType>() in the filter convention config fixes things.

### Steps to reproduce

1. Pascal also mentioned he was able to reproduce it so I am guessing not adding good steps here is fine :D

### Product

Hot Chocolate

### Version

13.0.0.0-preview.28

**Comment by glen-84** [comment]
@PascalSenn `uint` is a struct type, is this a duplicate of #4467?

**Comment by PascalSenn** [comment]
So there is a bit of a history behind this issue.

In the past we were generating all input types for filtering. This lead to awful type names like Int16FilterOperationInpuType. We then decided we will just fully type them out and bind them directly to the proper type.

Therefor, when you use e.g. UnsignedInt, then you will have to create a custom filter input type for this. I updated the documentation in #5759
